command/harness/setup.go

```diff
@@ -96,17 +93,17 @@ func HandleSetup(ctx context.Context, r *SetupVMRequest, s store.StageOwnerStore
 	if !poolManager.Exists(pool) {
 		return nil, fmt.Errorf("pool not defined")
 	}
-
-	instance, err := poolManager.Provision(ctx, pool, env.Runner.Name, env.Settings.LiteEnginePath)
+	_, err := s.Find(ctx, stageRuntimeID, pool)
 	if err != nil {
-		return nil, fmt.Errorf("failed provisioning")
+		if err = s.Create(ctx, &types.StageOwner{StageID: stageRuntimeID, PoolName: pool}); err != nil {
+			return nil, fmt.Errorf("could not create stage owner entity: %w", err)
+		}
 	}
 
-	if instance.IsHibernated {
-		instance, err = poolManager.StartInstance(ctx, pool, instance.ID)
-		if err != nil {
-			return nil, fmt.Errorf("failed to start the instance up")
-		}
+	instance, err := poolManager.Provision(ctx, pool, env.Runner.Name, env.Settings.LiteEnginePath)
+	if err != nil {
+		logr.Errorln("failed to provision instance")
```

Comment:
Can we print the error here as well: `logr.WithError(err).Errorln("failed to provision instance")`